src/less/button/button.less

```diff
@@ -212,15 +212,15 @@ button.btn--secondary:not([disabled], [aria-disabled="true"]),
 a.fake-btn--secondary[href] {
     &:active {
         background-color: var(--color-state-primary-active);
-        border-color: var(--color-state-button-accent-active);
-        color: var(--color-state-button-accent-active);
+        border-color: var(--color-state-active-foreground-on-secondary);
```

Comment:
I realize the border points to the same color as foreground, but it's very odd to have a prop of `...foreground...` being applied to `border-color`. The same thing for `destructive`. My suggestion was to have distinct props of `stroke` and `foreground`. If we really want to use a single prop for both `border` and `color`, we should probably use something more generic than `foreground` since that is typically assigned to `color`, not `border-color`.

I might be a bit too nitpicky; it just feels a but uneasy. @ianmcburnie , thoughts?

Reply:
Not nitpicky. It's a valid comment. I went with foreground for anything that picks up 'currentColor' in the CSS cascade (like border does). That was the reason anyway. It could feel pretty flimsy now... definitely open to changing it if feel it is confusing.

Reply:
Gotcha. Ok. Yea, I think it's preferable to be explicit here rather than depending on `currentColor` which may or may not apply. In the case of `state` props it's probably not going to apply that well since the existence of the state props themselves implies a difference in styling from the rest state.

Reply:
I think it's probably better to be more explicit, but lets keep it for now since this is in line with the rest of the variables. 
Next major we should revisit all of them, rather than fork a couple off for now.

Reply:
Sounds good.